packages/tiny-router/src/actions.ts

```diff
@@ -43,16 +44,24 @@ export const init: TinyRouter["actions"]["router"]["init"] = ({
   }
 };
 
-export const beforeSSR: TinyRouter["actions"]["router"]["beforeSSR"] = async ({
+export const beforeSSR: TinyRouter["actions"]["router"]["beforeSSR"] = ({
   state,
   actions
-}) => {
+}) => async (
+  ctx: Koa.ParameterizedContext<Koa.DefaultState, Koa.DefaultContext>
+) => {
   if (state.router.autoFetch) {
-    if (actions.source && actions.source.fetch)
+    if (actions.source && actions.source.fetch) {
       await actions.source.fetch(state.router.link);
-    else
+    }
+
+    const data = state.source.get(state.router.link);
+    if (data.isError) {
+      ctx.status = data.errorStatus;
```

Comment:
Hey @michalczaplinski, I guess you're solving this issue in this PR as well, right? https://github.com/frontity/frontity/issues/264

If so, could you please link the issue, assign it to you and move it to in progress? Thanks!

Reply:
yup, done.